### ISSUE TITLE
SD/SU sequences don't obey DEC scrolling regions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dragonlinux/connectbot#583]
```
What steps will reproduce the problem?

The SD/SU CSI sequences (CSI S, CSI T) do not obey the DEC scrolling regions 
(CSI ? r).

Find attached a text file. cat it on a terminal


What is the expected output? What do you see instead?

It should appear like the attached screenshots of xterm or pangoterm.

Instead it appears as also attached a screenshot of ConnectBot.


What version of the product are you using (you can see this by using Menu
-> About in the Host List)?

1.7.1

```

Original issue reported on code.google.com by `pev...@google.com` on 8 Jun 2012 at 1:27

Attachments:
- [scroll-test.txt](https://storage.googleapis.com/google-code-attachments/connectbot/issue-583/comment-0/scroll-test.txt)
- [scroll-xterm.png](https://storage.googleapis.com/google-code-attachments/connectbot/issue-583/comment-0/scroll-xterm.png)
- [scroll-pangoterm.png](https://storage.googleapis.com/google-code-attachments/connectbot/issue-583/comment-0/scroll-pangoterm.png)
- [scroll-connectbot.png](https://storage.googleapis.com/google-code-attachments/connectbot/issue-583/comment-0/scroll-connectbot.png)

**Comment by GoogleCodeExporter** [comment]
Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 1:32
- Added labels: **Component-Terminal**

**Comment by GoogleCodeExporter** [comment]
Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 1:34
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 3860493c8615.
```

Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 2:04
- Changed state: **Fixed**
